sdk/src/main/kotlin/io/rover/sdk/ui/blocks/poll/ViewTextPoll.kt

```diff
@@ -1,47 +1,116 @@
-        binding?.viewModel?.textPollBlock?.let {
-            createViews(it)
+
+    companion object {
+        private const val MAX_OPTIONS_AMOUNT = 4
+    }
+
+    private val optionIds = (0 until MAX_OPTIONS_AMOUNT).map { ViewCompat.generateViewId() }
+    private val questionId = ViewCompat.generateViewId()
+
+    override var viewModelBinding: MeasuredBindableView.Binding<TextPollViewModelInterface>? by ViewModelBinding { binding, subscriptionCallback ->
+        binding?.viewModel?.let { viewModel ->
+            if (view.childCount == 0) {
+                view.addView {
+                    view.textView {
+                        id = questionId
+                        setupLayoutParams(width = ViewGroup.LayoutParams.MATCH_PARENT, height = ViewGroup.LayoutParams.WRAP_CONTENT)
+                    }
+                }
+            }
+
+            bindQuestion(viewModel.textPollBlock)
+            setupOptionViews(viewModel)
+            informOptionBackgroundAboutSize(viewModel)
+
+            viewModel.votingState.androidLifecycleDispose(view).subscribe ({ votingState ->
+                when (votingState) {
+                    is VotingState.WaitingForVote -> { }
+                    is VotingState.Results -> setVoteResultsReceived(votingState)
+                }
+            }, { throw (it) }, { subscriptionCallback(it) })
         }
     }
 
-    private fun createViews(textPollBlock: TextPollBlock) {
-        val question = view.textView {
-            text = textPollBlock.question
+    private fun setupOptionViews(viewModel: TextPollViewModelInterface) {
+        val optionViews = createOptionViews(viewModel.textPollBlock)
+        startListeningForOptionImageUpdates(viewModel.optionBackgroundViewModel, optionViews)
+        optionViews.forEachIndexed { index, optionView ->
+            view.addView(optionView)
+            optionView.setOnClickListener { viewModelBinding?.viewModel?.castVote(index) }
+        }
+    }
+
+    private fun informOptionBackgroundAboutSize(viewModel: TextPollViewModelInterface) {
+        viewModelBinding?.measuredSize?.width?.let { measuredWidth ->
+            val optionStyleHeight = viewModel.textPollBlock.optionStyle.height.toFloat()
+            val measuredSize = MeasuredSize(measuredWidth, optionStyleHeight, view.resources.displayMetrics.density)
+            viewModel.optionBackgroundViewModel.informDimensions(measuredSize)
         }
+    }
+
+    private fun setVoteResultsReceived(votingResults: VotingState.Results) {
+        votingResults.votingShare.forEachIndexed { index, votingShare ->
+            val option = view.findViewById<TextOptionView>(optionIds[index])
+            val isSelectedOption = index == votingResults.selectedOption
+            viewModelBinding?.viewModel?.let {
+                option.goToResultsState(votingShare, isSelectedOption, it.textPollBlock.optionStyle)
```

Comment:
Hm, one thing I notice is that because we are skipping having an Option view model that it means we are sort of treating this as an old-school Presenter and calling a method on the View (well, the view mixin in this case) rather than having the View subscribe to it. I suppose that's okay for now though, I don't think we can really elegantly avoid it as long as we skip the view model.